server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -862,9 +861,11 @@ public void deleteStaleSegments(int lastNMetadataFilesToKeep) throws IOException
 
         Tuple<Long, Set<Long>> pinnedTimestampsState = RemoteStorePinnedTimestampService.getPinnedTimestamps();
 
+        Set<Long> pinnedTimestamps = new HashSet<>(pinnedTimestampsState.v2());
```

Comment:
(Readability) As someone new to this code, I'm not sure this variable naming clearly conveys what's happening. It's entirely possible it does, and the old naming with "State.v2()" is the confusing part.  TLDR: consider a comment here explaining what we're doing.

Reply:
We will be adding extra set of integ tests in a separate PR, let me address this as part of that PR.